src/crimson/os/seastore/journal.cc

```diff
@@ -404,28 +404,34 @@ Journal::find_replay_segments_fut Journal::find_replay_segments()
     });
 }
 
-Journal::read_record_metadata_ret Journal::read_record_metadata(
-  paddr_t start)
+Journal::read_validate_record_metadata_ret Journal::read_validate_record_metadata(
+  paddr_t start,
+  segment_nonce_t nonce)
 {
   if (start.offset + block_size > (int64_t)segment_manager.get_segment_size()) {
-    return read_record_metadata_ret(
-      read_record_metadata_ertr::ready_future_marker{},
+    return read_validate_record_metadata_ret(
+      read_validate_record_metadata_ertr::ready_future_marker{},
```

Comment:
nit, FWIW, could just put
```c++
return {read_validate_record_metadata_ertr::ready_future_marker{},
        std::nullopt};
```
as C++17 is able to perform auto deduction from braced-init-list.

Reply:
There are actually a lot of lambdas in this file where the auto-deduction fails because there are multiple returns.  I'll keep this in mind in the future, but I'm not going to bother with it right now.